File: snapshots/input/syntax/Main/Methods.cs
using System.Diagnostics.CodeAnalysis;

namespace Main;

[SuppressMessage("ReSharper", "all")]
public class Methods
{
    int SingleParameter(int b)
    {
        return b;
    }

    int TwoParameters(int a, int b)
    {
        return a + b;
    }

    int Overload1(int a)
    {
        return a;
    }

    int Overload1(int a, int b)
    {
        return a + b;
    }

    T Generic<T>(T param)
    {
        return param;
    }

    T GenericConstraint<T>(T param) where T : new()
    {
        return param;
    }

    void DefaultParameter(int a = 5)
    {
    }

    int DefaultParameterOverload(int a = 5)
    {
        return DefaultParameterOverload(a, a);
    }

    int DefaultParameterOverload(int a, int b)
    {
        return DefaultParameterOverload();
    }

    interface IHello
    {
        string Hello();
    }

    class ImplementsHello : IHello
    {
        string IHello.Hello()
        {
            return "";
        }
    }

    class InheritedOverloads1
    {
        public void Method()
        {
        }
    }

    class InheritedOverloads2 : InheritedOverloads1
    {
        public int Method(int parameter)
        {
            return parameter;
        }
    }

    class InheritedOverloads3 : InheritedOverloads2
    {
        public string Method(string parameter)
        {
            return parameter;
        }
    }

    public static void InheritedOverloads()
    {
        new InheritedOverloads1().Method();
        new InheritedOverloads2().Method();
        new InheritedOverloads2().Method(42);
        new InheritedOverloads3().Method();
        new InheritedOverloads3().Method(42);
        new InheritedOverloads3().Method("42");
    }

}

File: snapshots/input/syntax/VBMain/Methods.vb
Imports System.Diagnostics.CodeAnalysis

Namespace VBMain
    <SuppressMessage("ReSharper", "all")>
    Public Class Methods
        Private Function SingleParameter(ByVal b As Integer) As Integer
            Return b
        End Function

        Private Function TwoParameters(ByVal a As Integer, ByVal b As Integer) As Integer
            Return a + b
        End Function

        Private Function Overload1(ByVal a As Integer) As Integer
            Return a
        End Function

        Private Function Overload1(ByVal a As Integer, ByVal b As Integer) As Integer
            Return a + b
        End Function

        Private Function Generic(Of T)(ByVal param As T) As T
            Return param
        End Function

        Private Function GenericConstraint(Of T As New)(ByVal param As T) As T
            Return param
        End Function

        Private Sub DefaultParameter(ByVal Optional a As Integer = 5)
        End Sub

        Private Function DefaultParameterOverload(ByVal Optional a As Integer = 5) As Integer
            Return DefaultParameterOverload(a, a)
        End Function

        Private Function DefaultParameterOverload(ByVal a As Integer, ByVal b As Integer) As Integer
            Return DefaultParameterOverload()
        End Function

        Interface IHello
            Function Hello() As String
        End Interface

        Class ImplementsHello
            Implements IHello

            Private Function Hello() As String Implements IHello.Hello
                Throw New NotImplementedException()
            End Function

        End Class

        Class InheritedOverloads1
            Public Sub Method()
            End Sub
        End Class

        Class InheritedOverloads2
            Inherits InheritedOverloads1

            Public Function Method(ByVal parameter As Integer) As Integer
                Return parameter
            End Function
        End Class

        Class InheritedOverloads3
            Inherits InheritedOverloads2

            Public Function Method(ByVal parameter As String) As String
                Return parameter
            End Function
        End Class

        Public Shared Sub InheritedOverloads()
            Dim a As InheritedOverloads1 = New InheritedOverloads1
            a.Method()
            Dim b As InheritedOverloads2 = New InheritedOverloads2
            DirectCast(b, InheritedOverloads1).Method()
            b.Method(42)
            Dim c As InheritedOverloads3 = New InheritedOverloads3
            DirectCast(c, InheritedOverloads1).Method()
            DirectCast(c, InheritedOverloads2).Method(42)
            c.Method("42")
        End Sub
    End Class
End Namespace
